src/components/geneOntologyRibbon/index.js

```diff
@@ -1,61 +1,475 @@
+import ControlsContainer from '../controlsContainer';
+import { selectOrthologs } from '../../selectors/geneSelectors';
+import OrthologPicker from '../OrthologPicker';
+import { connect } from 'react-redux';
+import { getOrthologId } from '../orthology';
+
+const goApiUrl = 'http://api.geneontology.org/api/';
+
+const exp_codes = ['EXP', 'IDA', 'IPI', 'IMP', 'IGI', 'IEP', 'HTP', 'HDA', 'HMP', 'HGI', 'HEP'];
+
+const Checkbox = props => (
+  <input type="checkbox" {...props} />
+);
 
 class GeneOntologyRibbon extends Component {
 
-  static hasBlockData(entities){
-    for (let entity of entities) {
-      if(entity.blocks && entity.blocks.length > 0) {
-        return true;
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading : true,
+      subjectBaseURL : '/gene/',
+      stringency: STRINGENCY_HIGH,
+      selectedOrthologs: [],
+      excludePB : true,
+      excludeIBA : true,
+      // onlyEXPWhenMulti : true,
+      onlyEXP : false,
+      selected : {
+        subject : null,
+        group : null,
+        data : null,
+        ready : false,
+      },
+      search : ''
+    };
+    this.handleOrthologyChange = this.handleOrthologyChange.bind(this);
+  }
+
+
+  getGeneIdList() {
+    return [this.props.id].concat(this.state.selectedOrthologs.map(getOrthologId));
+  }
+
+  handleOrthologyChange(selectedOrthologs) {
+    this.setState({selectedOrthologs}, () => {
+      this.fetchData('goslim_agr', this.getGeneIdList()).then(data => {
+        var oldSubs = [];
+        if(this.state.ribbon) {
+          oldSubs = this.state.ribbon.subjects;
+        }
+        for(var sub of data.data.subjects) {
+          oldSubs.push(sub);
+        }
+        this.setState({ loading : false, ribbon : data.data, subjects : oldSubs,selected : {
+          subject : null,
+          group : null,
+          data : null,
+          ready : false,
+        } });
+      });
+    });
+  }
+
+  ribbonOptions(subjects) {
+    var excludeIBA = this.state.excludeIBA && subjects.length > 1;
+    var exps = '';
+    if(this.state.onlyEXP) {
+      for(var exp of exp_codes) {
+        exps += '&ecodes=' + exp;
+      }
+    }
+    return '&exclude_PB=' + this.state.excludePB + '&exclude_IBA=' + excludeIBA + exps;
+  }
+
+  fetchData(subset, subjects) {
+    var subs = '';
+    if(subjects instanceof Array) {
+      subs = subjects.join('&subject=');
+    }
+    let query = goApiUrl + 'ontology/ribbon/?subset=' + subset + '&subject=' + subs + this.ribbonOptions(subjects);
+    // console.log('Query is ' + query);
+    return axios.get(query);
+  }
+
+  fetchAssociationData(subject, group) {
+    if(group == 'all') {
+      var groups = this.state.ribbon.categories.map(elt => {
+        return elt.id;
+      });
+      group = groups.join('&slim=');
+    }
+    let query = goApiUrl + 'bioentityset/slimmer/function?slim=' + group + '&subject=' + subject + '&rows=-1';
+    return axios.get(query);
```

Comment:
`axios` is not a listed dependency of the agr_ui project. I don't like the idea of relying on dependencies of dependencies. Any reason to not use the `fetchData` function un `src/lib/fetchData.js`? 

I guess a bigger question is is there a reason that this is making API calls directly in the component, as opposed to going through redux?